### PR TITLE
Repeat question and ask for another attempt if user choice is not valid

### PR DESCRIPTION
**Pull request by passionfruitjello**

### CODE REVIEW

File: src/main/java/io/OutputHandlerImpl.java [hunk 1]
```diff
@@ -19,12 +21,27 @@ public static OutputHandlerImpl getScreen() {
         return screen;
     }
 
+    /**
+     * @return the last question
+     */
+    public String getText() {
```
Comment (bw55555) [inline src/main/java/io/OutputHandlerImpl.java:27]: be more descriptive with the name
like getLastText()?

**Review by bw55555 — CHANGES_REQUESTED** [review]
Should add the methods to OutputHandler interface

File: src/main/java/io/InputHandlerImpl.java [hunk 2]
```diff
@@ -28,7 +29,15 @@ public String getChoice(InputValidator validator) {
             OutputHandlerImpl screen = OutputHandlerImpl.getScreen();
             String msg = validator.getErrorMessage(choice);
             screen.generateText(msg);
-            screen.generateText("Your choice is not valid. Please attempt your choice again.");
+            screen.generateText("Your choice is not valid. Please attempt again.");
+            String text = screen.getLastText();
+            List<String> options = screen.getLastOptions();
+            if (options == null) {
```
Comment (bw55555) [inline src/main/java/io/InputHandlerImpl.java:35]: this logic doesn't work unless you overwrite options in generateText too

**Review by bw55555 — CHANGES_REQUESTED** [review]
minor bug

**Review by bw55555 — APPROVED** [review]
looks good

**Review by Yasutakahi — APPROVED** [review]
I think in terms of the cleanness of the code, it is very nice as it is following the clean architecture and SOLID principles. However you might have to tests for these class so you can make sure the code works well in the future.

**Review by mahirhassan-uoft — APPROVED** [review]
cool

**Pull request merged**